src/scripts/main.js

```diff
@@ -1,3 +1,315 @@
+  { value: 'London', text: 'London' },
+  { value: 'New York', text: 'New York' },
+  { value: 'Edinburgh', text: 'Edinburgh' },
+  { value: 'San Francisco', text: 'San Francisco' },
+];
+
+select.forEach((selectData) => {
+  const option = document.createElement('option');
+
+  option.value = selectData.value;
+  option.textContent = selectData.text;
+  inputOffice.append(option);
+});
+
+labelOffice.appendChild(inputOffice);
+form.append(labelOffice);
+
+const inputAge = document.createElement('input');
+const labelAge = document.createElement('label');
+
+inputAge.type = 'number';
+inputAge.setAttribute('data-qa', 'age');
+inputAge.setAttribute('required', '');
+labelAge.textContent = 'Age:';
+labelAge.appendChild(inputAge);
+form.append(labelAge);
+
+const inputSalary = document.createElement('input');
+const labelSalary = document.createElement('label');
+
+inputSalary.type = 'number';
+inputSalary.setAttribute('data-qa', 'salary');
+inputSalary.setAttribute('required', '');
+labelSalary.textContent = 'Salary:';
+labelSalary.appendChild(inputSalary);
+form.append(labelSalary);
+
+const button = document.createElement('button');
+
+button.type = 'submit';
+button.textContent = 'Save to table';
+form.append(button);
+
+document.body.append(form);
+
+button.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const existingNotification = document.querySelector('.notification');
+
+  if (existingNotification) {
+    existingNotification.remove();
+  }
+
+  const notification = document.createElement('div');
+
+  notification.className = 'notification';
+
+  notification.setAttribute('data-qa', 'notification');
+
+  let isValid = true;
+
+  if (
+    inputName.value.trim().length < 4 ||
+    !/^[a-zA-Z\s]+$/.test(inputName.value)
+  ) {
+    isValid = false;
+    notification.classList.add('error');
+
+    notification.textContent =
+      'Error! Name value should have more than 4 letters!';
+    document.body.append(notification);
+  }
+
+  if (inputAge.value < 18 || inputAge.value > 90) {
+    isValid = false;
+    notification.classList.add('error');
+
+    notification.textContent =
+      'Error! Age value should be more than 18 and less than 90';
+    document.body.append(notification);
+  }
+
+  if (!isValid) {
+    e.preventDefault();
+
+    return;
+  }
+
+  const additionalRow = document.createElement('tr');
+
+  for (let i = 0; i <= 4; i++) {
+    const additionalCell = document.createElement('td');
+
+    additionalRow.append(additionalCell);
+  }
+
+  additionalRow.cells[0].textContent = inputName.value;
+  additionalRow.cells[1].textContent = inputPosition.value;
+  additionalRow.cells[2].textContent = inputOffice.value;
+  additionalRow.cells[3].textContent = inputAge.value;
+
+  const modifiedSalary = () => {
+    if (inputSalary.value.length <= 3) {
+      return '$' + inputSalary.value;
+    }
+
+    return (
+      '$' + inputSalary.value.slice(0, -3) + ',' + inputSalary.value.slice(-3)
+    );
```

Comment:
The `modifiedSalary` function only formats salaries with up to three digits. Consider enhancing this logic to handle larger numbers correctly, ensuring proper comma placement for thousands, millions, etc.